dbms/src/Storages/StorageHDFS.cpp

```diff
@@ -0,0 +1,178 @@
+#include <Storages/StorageFactory.h>
+#include <Storages/StorageHDFS.h>
+#include <Interpreters/Context.h>
+#include <Interpreters/evaluateConstantExpression.h>
+#include <Parsers/ASTIdentifier.h>
+#include <Parsers/ASTLiteral.h>
+#include <IO/ReadBufferFromHDFS.h>
+#include <Formats/FormatFactory.h>
+#include <DataStreams/IBlockOutputStream.h>
+#include <DataStreams/UnionBlockInputStream.h>
+#include <DataStreams/IProfilingBlockInputStream.h>
+#include <DataStreams/OwningBlockInputStream.h>
+#include <Poco/Path.h>
+#include <TableFunctions/ITableFunction.h>
+#include <Common/typeid_cast.h>
+
+namespace DB
+{
+namespace ErrorCodes
+{
+    extern const int NUMBER_OF_ARGUMENTS_DOESNT_MATCH;
+}
+
+StorageHDFS::StorageHDFS(const String & uri_,
+    const std::string & table_name_,
+    const String & format_name_,
+    const ColumnsDescription & columns_,
+    Context & context_)
+    : IStorage(columns_), uri(uri_), format_name(format_name_), table_name(table_name_), context_global(context_)
+{
+}
+
+namespace
+{
+    class StorageHDFSBlockInputStream : public IProfilingBlockInputStream
+    {
+    public:
+        StorageHDFSBlockInputStream(const String & uri,
+            const String & format,
+            const String & name_,
+            const Block & sample_block,
+            const Context & context,
+            size_t max_block_size)
+            : name(name_)
+        {
+            // Assume no query and fragment in uri, todo, add sanity check
+            String fuzzyFileNames;
+            String uriPrefix = uri.substr(0, uri.find_last_of('/'));
+            if (uriPrefix.length() == uri.length())
+            {
+                fuzzyFileNames = uri;
+                uriPrefix.clear();
+            }
+            else
+            {
+                uriPrefix += "/";
+                fuzzyFileNames = uri.substr(uriPrefix.length());
+            }
+
+            std::vector<String> fuzzyNameList = parseDescription(fuzzyFileNames, 0, fuzzyFileNames.length(), ',' , 100/* hard coded max files */);
```

Comment:
They are the same :)

Reply:
Let's leave only the first. Because it is compatible with the syntax of supported "globs" in bash.
And we can eventually add the support for globs in StorageFile.